src/hmc.py

```diff
@@ -0,0 +1,298 @@
+import logging
+
+from memory import Memory
+
+LOGGER = logging.getLogger(__name__)
+
+class Hmc(Memory):
+    """
+    General HMC memory on the FPGA.
+    """
+    def __init__(self, parent, device_name, address, length_bytes, mezzanine_site, device_info=None, legacy_reg_map=True):
+        super(Hmc, self).__init__(name=device_name, width_bits=256, address=address, length_bytes=length_bytes)
+        self.parent = parent
+        self.device_info = device_info
+        self.device_name = device_name
+        self.mezz_site = mezzanine_site
+        self.reg_map = {# HMC Status Registers (LINK2)
```

Comment:
Can the reg_map be moved to skarab_definitions.py?

Reply:
Just a thought, maybe skarab_definitions.py isn't the best place for it? There could, for argument's sake, be 
SKARABs without HMCs. We'd rather, then, want to contain all HMC related info.

Reply:
I am going to leave as is. I have discussed this with Wesley and he suggested that it is better practice that this gets moved to hmc.py. The whole idea of skarab_definitions.py was to accommodate the SKARAB BSP functions. The HMC is part of the DSP toolflow. Tyrone raises a valid point.